src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -135,14 +135,19 @@ internal void ReleaseConnection()
                 {
                     if (!_isClosed && _tcpClient != null)
                     {
-                        //free cbt buffer
-                        if (_channelBindingToken != null)
+                        try
                         {
-                            _channelBindingToken.Close();
+                            // Gracefully close the transmission channel
+                            _tcpClient.Client.Blocking = false;
+                            QuitCommand.Send(this);
```

Comment:
I think this can end up getting called if we're in the middle of a send and end up with an abortive `Dispose`; this would result in sending `QUIT` as part of message data. We should verify that this is not the case or add a check to see if this is clean vs abortive.

Reply:
@MarcoRossignoli can you rebase on master (some changes were made in Abort), and merge the methods?

Reply:
yep sorry for delay, was out sick.

Reply:
Merged